packages/flutter/lib/src/services/sensitive_content.dart

```diff
@@ -0,0 +1,151 @@
+  ///
+  /// This mode cannot be used to set the sensitivity level of a `SensitiveContent` widget.
+  _unknown,
+}
+
+/// Service for setting the content sensitivity of the native app window (Android `View`)
+/// that contains the app's widget tree.
+///
+/// This service is only currently supported on Android API 35+.
+class SensitiveContentService {
+  /// Creates service to set content sensitivity of an app window (Android `View`) via
+  /// communication over the sensitive content [MethodChannel].
+  SensitiveContentService() {
+    sensitiveContentChannel = SystemChannels.sensitiveContent;
+  }
+
+  /// The channel used to communicate with the shell side to get and set the
+  /// content sensitivity of an app window (Android `View`).
+  late MethodChannel sensitiveContentChannel;
+
+  /// Sets content sensitivity level of the app window (Android `View`) that contains the app's widget
+  /// tree to the level specified by [contentSensitivity] via a call to the native embedder.
+  Future<void> setContentSensitivity(ContentSensitivity contentSensitivity) async {
+    await sensitiveContentChannel.invokeMethod<void>(
+      'SensitiveContent.setContentSensitivity',
+      contentSensitivity.index,
+    );
+  }
+
+  /// Gets content sensitivity level of the app window (Android `View`) that contains
+  /// the app's widget tree.
+  Future<ContentSensitivity> getContentSensitivity() async {
+    final int? result = await sensitiveContentChannel.invokeMethod<int>(
+      'SensitiveContent.getContentSensitivity',
+    );
+
+    final ContentSensitivity contentSensitivity = ContentSensitivity.values.firstWhere(
+      (ContentSensitivity cs) => cs.index == result && cs != ContentSensitivity._unknown,
+      orElse:
+          () =>
+              throw UnsupportedError(
+                'Android Flutter View has a content sensitivity mode '
+                'that is not recognized by Flutter. If you see this error, it '
+                'is possible that the View uses a new mode that Flutter needs to '
+                'support; please file an issue.',
+              ),
+    );
+
+    return contentSensitivity;
+  }
+
+  /// Returns whether or not setting content sensitivity levels is supported
+  /// by the device.
+  ///
+  /// This method must be called before attempting to call [getContentSensitivity]
+  /// or [setContentSensitivity].
+  ///
+  /// This feature is only supported on Android 35+ currently. Its return value will
+  /// not change and thus, is safe to cache.
+  Future<bool> isSupported() async {
+    if (defaultTargetPlatform != TargetPlatform.android) {
+      return false;
+    }
+
+    try {
+      final bool result =
+          (await sensitiveContentChannel.invokeMethod<bool>('SensitiveContent.isSupported'))!;
+      return result;
+    } catch (e) {
+      // If this call fails or unexpectedly returns null, then consider setting content
+      // sensitivity unsupported.
+      return false;
+    }
```

Comment:
This method looks like it will swallow errors and app developers might not be aware that something went wrong. They probably do want to assume that an error or null means `false`, but in addition I would probably do FlutterError.reportError.

You may want to do the error handling at the level of the caller of this method in order to align with how error handling is done for the other methods in this file, but it's up to you. The important part is just that errors don't get swallowed.